src/processors/SwaggerInputProcessor.ts

```diff
@@ -0,0 +1,109 @@
+import { AbstractInputProcessor } from './AbstractInputProcessor';
+import { JsonSchemaInputProcessor } from './JsonSchemaInputProcessor';
+import { CommonInputModel, SwaggerV2Schema } from '../models';
+import { Logger } from '../utils';
+import SwaggerParser from '@apidevtools/swagger-parser';
+import { OpenAPIV2 } from 'openapi-types';
+
+/**
+ * Class for processing Swagger inputs
+ */
+export class SwaggerInputProcessor extends AbstractInputProcessor {
+  static supportedVersions = ['2.0'];
+
+  /**
+   * Process the input as a Swagger document
+   * 
+   * @param input 
+   */
+  async process(input: Record<string, any>): Promise<CommonInputModel> {
+    if (!this.shouldProcess(input)) {throw new Error('Input is not a Swagger document so it cannot be processed.');}
+
+    Logger.debug('Processing input as a Swagger document');
+    const common = new CommonInputModel();
+    common.originalInput = input;
+    
+    //Since we require that all references have been dereferenced, we cannot "simply" support already parsed inputs.
+    const api = (await SwaggerParser.dereference(input as any) as unknown) as OpenAPIV2.Document;
+    for (const [path, pathObject] of Object.entries(api.paths)) {
+      //Remove all special chars from path
+      let formattedPathName = path.replace(/[^\w\s*]+/g, '');
+      //Remove any pre-pending '/'
+      formattedPathName = formattedPathName.replace(/\//, '');
+      //Replace all segment separators '/'
+      formattedPathName = formattedPathName.replace(/\//gm, '_');
+      this.processOperation(pathObject.get, `${formattedPathName}_get`, common);
+      this.processOperation(pathObject.put, `${formattedPathName}_put`, common);
+      this.processOperation(pathObject.post, `${formattedPathName}_post`, common);
+      this.processOperation(pathObject.options, `${formattedPathName}_options`, common);
+      this.processOperation(pathObject.head, `${formattedPathName}_head`, common);
+      this.processOperation(pathObject.patch, `${formattedPathName}_patch`, common);
```

Comment:
I don't know so I ask. Does Swagger also have something like components, where you can have reusable definitions for parameters and response?

Reply:
Also, maybe you should use as fallback the `id` field for model name?

Reply:
They have something called `definitions` rather than `components`. Do you think we should include such a "complex" id reflection here? 🤔 

I was thinking it was best with a basic implementation, at least for now.

Reply:
> I was thinking it was best with a basic implementation, at least for now.

If you wanna go with next PRs, current solution is good for me :)

Reply:
@magicmatatjahu if the swagger document has lonely definitions (that are not referenced anywhere), the data models will not be generated - and I am not planning to add any PR after this one, this PR is more a primer to include more users that can use it.